catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/MetacardMarshallerImpl.java

```diff
@@ -179,75 +181,82 @@ public String marshal(Metacard metacard, Map<String, Serializable> arguments)
         return writer.makeString();
     }
 
+    private String getStringValue(XmlPullParser parser, Attribute attribute,
+            AttributeType.AttributeFormat format, Serializable value)
+            throws IOException, CatalogTransformerException {
+        switch (format) {
+
+        case STRING:
+        case BOOLEAN:
+        case SHORT:
+        case INTEGER:
+        case LONG:
+        case FLOAT:
+        case DOUBLE:
+            return value.toString();
+        case DATE:
+            Date date = (Date) value;
+            return DateFormatUtils.formatUTC(date, DF_PATTERN);
+        case GEOMETRY:
+            return geoToXml(geometryTransformer.transform(attribute), parser);
+        case OBJECT:
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            try (ObjectOutput output = new ObjectOutputStream(bos)) {
+                output.writeObject(attribute.getValue());
+                return Base64.getEncoder()
+                        .encodeToString(bos.toByteArray());
+            }
+        case BINARY:
+            return Base64.getEncoder()
+                    .encodeToString((byte[]) value);
+        case XML:
+            return value.toString()
+                    .replaceAll("[<][?]xml.*[?][>]", "");
+        default:
+            LOGGER.warn("Unsupported attribute: {}", format);
+            return value.toString();
+        }
+    }
+
     private void writeAttributeToXml(PrintWriter writer, XmlPullParser parser, Attribute attribute,
             AttributeType.AttributeFormat format) throws IOException, CatalogTransformerException {
         String attributeName = attribute.getName();
 
+        List<String> values = new ArrayList<>();
```

Comment:
Don't know if it matters much, but looks like you could avoid creating this list by doing your length check on `attribute.getValues()`, then write the outer xml on either side of the first `forEach`.

If you do keep this list, you could also give it a more precise name like `stringifiedValues` or something.